OrbitCaptureClient/CaptureClient.cpp

```diff
@@ -85,14 +106,38 @@ void CaptureClient::Capture(
   FinishCapture();
 }
 
-void CaptureClient::StopCapture() {
+bool CaptureClient::StopCapture() {
+  absl::MutexLock lock(&state_mutex_);
+  if (state_ == State::kStarting) {
+    // Block and wait until the state is not kStarting
+    bool (*IsNotStarting)(State * state) = [](State* state) {
+      return *state != State::kStarting;
+    };
+    state_mutex_.Await(absl::Condition(IsNotStarting, &state_));
+  }
+
+  if (state_ != State::kStarted) {
+    LOG("StopCapture ignored, because it is already stopping or stopped");
+    return false;
+  }
+
   CHECK(reader_writer_ != nullptr);
 
   if (!reader_writer_->WritesDone()) {
-    ERROR("Finishing writing on Capture's gRPC stream");
-    FinishCapture();
+    // Normally the capture thread waits until service stops sending messages,
+    // but in this case since we failed to notify the service we pull emergency
+    // stop plug. Setting this flag forces capture thread to exit as soon
+    // as it notices that it was set.
+    ERROR(
+        "WritesDone failed unable to finish the capture in orderly manner, "
```

Comment:
```suggestion
        "WritesDone on Capture's gRPC stream failed: unable to finish the capture in orderly manner, "
```
(with the proper clang-format)

Reply:
done.